client/extensions/woocommerce/components/custom-tree-selector/custom-tree-node.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+import FormLabel from 'components/forms/form-label';
+import FormInputCheckbox from 'components/forms/form-checkbox';
+
+const CustomTreeNode = ( {
+	node,
+	onNodeSelect,
+} ) => {
+	const children = ( node.children ? renderChildren( node.children, onNodeSelect ) : null );
+
+	return (
+		<div className="custom-tree-selector__node">
+			<FormLabel>
```

Comment:
This label ends up wrapping all of the children, which includes other label elements, which is not valid HTML and would probably have a negative impact on accessibility. You could wrap it just around the renderSelector & span, and leave children outside it.

Actually, it also ends up causing some strange click behavior where the label for the parent element is clickable in spaces around/between children: https://cloudup.com/cn207kuISZG

Reply:
Ooh, good point. That is weird to be able to click in the margin like that.